code/game/objects/machinery/floodlight.dm

```diff
@@ -235,3 +235,100 @@
 	toggle_lights(turned_on ? FALSE : TRUE)
 	update_icon()
 	return TRUE
+
+// Hybrisa Streetlights
+/obj/machinery/floodlight/colony/street
+	name = "Colony Streetlight"
+	icon = 'icons/obj/structures/prop/urban/64x64_urbanrandomprops.dmi'
+	icon_state = "street_off"
+	layer = ABOVE_LYING_MOB_LAYER
+	brightness_on = 5
+	resistance_flags = XENO_DAMAGEABLE
+	max_integrity = 220
+	density = FALSE
+
+/obj/machinery/floodlight/colony/street/update_icon_state()
+	var/area/street_light_area = get_area(src)
+	if(obj_integrity != initial(max_integrity))
+		icon_state = "street_dmg"
+		set_light(0)
+	else if(street_light_area.power_light)
+		icon_state = "street_on"
+		set_light(5, 5, COLOR_WHITE)
+	else
+		icon_state = "street_off"
+		set_light(0)
+
+// Traffic
+
+/obj/machinery/floodlight/colony/traffic
+	name = "traffic light"
+	desc = "A traffic light"
+	icon = 'icons/obj/structures/prop/urban/64x64_urbanrandomprops.dmi'
+	icon_state = "trafficlight"
+	bound_width = 32
+	bound_height = 32
+	density = FALSE
+	max_integrity = 200
+	layer = ABOVE_LYING_MOB_LAYER
+	resistance_flags = XENO_DAMAGEABLE
+	brightness_on = 10
+
+/obj/machinery/floodlight/colony/traffic/update_icon_state()
+	var/area/traffic_light_area = get_area(src)
+	if(obj_integrity != initial(max_integrity))
+		icon_state = "trafficlight_damaged"
+		set_light(0)
+	else if(traffic_light_area.power_light)
+		icon_state = "trafficlight_on"
+		set_light(2, 5, COLOR_WHITE)
+	else
+		icon_state = "trafficlight"
+		set_light(0)
+
+/obj/machinery/floodlight/colony/traffic_alt
+	name = "traffic light"
+	desc = "A traffic light"
+	icon = 'icons/obj/structures/prop/urban/64x64_urbanrandomprops.dmi'
+	icon_state = "trafficlight_alt"
+	bound_width = 32
+	bound_height = 32
+	density = FALSE
+	max_integrity = 200
+	layer = ABOVE_MOB_LAYER
+	resistance_flags = XENO_DAMAGEABLE
+
+/obj/machinery/floodlight/colony/traffic_alt/update_icon_state()
+	if(obj_integrity != initial(max_integrity))
+		icon_state = "trafficlight_alt_damaged"
+		set_light(0)
+	else if(light_on)
+		icon_state = "trafficlight_alt_on"
+		set_light(7, 5, COLOR_WHITE)
+	else
+		icon_state = "trafficlight_alt"
+		set_light(0)
```

Comment:
I'm certain I reviewed this part but apparently its missing, so RIP.
These need to either just follow normal floodlight behavior, or if not, fix up the logic and procs.

Reply:
Done